qcengine/programs/nwchem/harvester.py

```diff
@@ -614,11 +614,103 @@ def harvest_outfile_pass(outtext):
     if mobj:
         psivar["N ALPHA ELECTRONS"] = mobj.group(2)
         psivar["N BETA ELECTRONS"] = mobj.group(3)
+
+        if psivar["N ALPHA ELECTRONS"] == psivar["N BETA ELECTRONS"]:
+
+            # get HOMO and LUMO energy
+            mobj = re.search(
+                r"Vector"
+                + r"\s+"
+                + r"%d" % (psivar["N ALPHA ELECTRONS"])
+                + r"\s+"
+                + r"Occ="
+                + r".*"
+                + r"\s+"
+                + r"E="
+                + r"([+-]?\s?\d+[.]\d+)"
+                + r"[D]"
+                + r"([+-])"
+                + r"[0]"
+                + r"(\d+)",
+                outtext,
+                re.MULTILINE,
+            )
+            if mobj:
+                if mobj.group(2) == "+":
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["HOMO"] = np.array([round(lumo, 10)])
+                else:
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["HOMO"] = np.array([round(lumo, 10)])
+
+            mobj = re.search(
+                r"Vector"
+                + r"\s+"
+                + r"%d" % (psivar["N ALPHA ELECTRONS"] + 1)
+                + r"\s+"
+                + r"Occ="
+                + r".*"
+                + r"\s+"
+                + r"E="
+                + r"([+-]?\s?\d+[.]\d+)"
+                + r"[D]"
+                + r"([+-])"
+                + r"[0]"
+                + r"(\d+)",
+                outtext,
+                re.MULTILINE,
+            )
+            if mobj:
+                if mobj.group(2) == "+":
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["LUMO"] = np.array([round(lumo, 10)])
+                else:
+                    lumo = float(mobj.group(1)) * (10 ** (-1 * float(mobj.group(3))))
+                    psivar["LUMO"] = np.array([round(lumo, 10)])
+
     mobj = re.search(r"AO basis - number of functions:\s+(\d+)\s+number of shells:\s+(\d+)", outtext, re.MULTILINE)
     if mobj:
         psivar["N MO"] = mobj.group(2)
         psivar["N BASIS"] = mobj.group(1)
 
+    # Search for Center of charge
+    mobj = re.search(
+        r"Center of charge \(in au\) is the expansion point"
+        + r"\n"
+        + r"\s+"
+        + r"X\s+=\s+([+-]?\d+[.]\d+)"
+        + r"\s+"
+        + r"Y\s+=\s+([+-]?\d+[.]\d+)"
+        + r"\s+"
+        + r"Z\s+=\s+([+-]?\d+[.]\d+)",
```

Comment:
this is fine if purposeful, but I wanted to make sure you're aware of `NUMBER`, which can simplify reading regex blocks

Reply:
I wouldn't say it's purposeful, I can try changing it to simplify the code.  I wasn't aware of `NUMBER`

Reply:
thanks, it'll help readability. fine as a future pass.